Update requirements for 3.11

plettich opened the pull request:
Working on #4070

plettich committed:
Update dependencies for v3.11

plettich committed:
Add requirements for Python 3.12

Fix tests to run with latest requirements and Python 3.12.

plettich committed:
Remove Python 3.7 from configuration

We decided not to support Python 3.7 anymore in privacyIDEA 3.11.
Also fixed a failing tests for language selection.

plettich committed:
Fix failing CLI test

plettich committed:
Update requirements for test and documentation

- Remove lower version boundary from requirements in setup.py we don't
  check them anyway.
- The requirements for tests and docs now contain the complete set of
  requirements from the server, it is easier to compile.
- Update GitHub action since we have the complete set of requirements
  already in the test requirements.
- Add a check if the gpg binary is installed, otherwise skip the import
  test.

plettich committed:
Fix requirements for Python 3.8

henrik-falk commented on setup.py:
```diff
@@ -29,40 +29,39 @@ def get_file_list(file_path):
+    "google-auth",
+    "grpcio",
+    "huey[redis]",
+    "importlib_metadata; python_version=='3.7'",
```
python version 3.7 is no longer supported

plettich replied:
Done in 0091b57

henrik-falk reviewed (approved):
besides the minimal remark we found, everything looks good

plettich committed:
Remove Python 3.7 remnants

plettich committed:
Update documentation

- Remove notes on Python versions less than 3.7
- Remove notes on unsupported OS
